### PR TITLE
Add macro `do_readme_rmd()`

### LINKED ISSUE #65: Add pkg documentation macro
Is there already/could there be a `tic` example showing how to deploy README.md from Travis when there's a README.Rmd and how to use `roxygen2::document` similarly?

I'd use that in so many packages :joy:

### PR DESCRIPTION
**Pull request by pat-s**
fixes #65

### CODE REVIEW

File: R/macro-readme-rmd.R [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                          branch = "master",
+                          remote_url = NULL,
+                          commit_message = NULL,
+                          ssh_key_name = "id_rsa") {
+
+  #' @description
+  #' 1. [step_setup_ssh()] in the `"before_deploy"` to setup
+  #'    the upcoming deployment
+  get_stage("before_deploy") %>%
+    add_step(step_setup_ssh(name = ssh_key_name))
```
Comment (krlmlr) [inline R/macro-readme-rmd.R]:
```suggestion
    add_step(step_setup_ssh(name = !!enquo(ssh_key_name)))
```

File: R/macro-readme-rmd.R [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+#'
+#' @description
+#' `do_readme_rmd()` renders an R Markdown README and deploys
+#' the rendered README.md file to Github. It adds default steps to the
+#' `"before_deploy"` and `"deploy"` stages:
+#'
+#' @inheritParams step_setup_ssh
+#' @inheritParams step_setup_push_deploy
+#' @inheritParams step_do_push_deploy
+#' @param path,branch By default, this macro deploys the `"master"` branch
```
Comment (krlmlr) [inline R/macro-readme-rmd.R]: Is gh-pages really an alternative?

Comment (pat-s) [reply R/macro-readme-rmd.R]: Not really, thanks.

**Comment by pat-s** [comment]
@krlmlr 

- Added a section about the macro to the "Getting started" vignette
- added a lifecycle badge including the framework via `usethis::use_lifecycle()`

**Comment by pat-s** [comment]
@krlmlr can we merge this?

**Pull request merged**